examples/operator.rs

```diff
@@ -0,0 +1,100 @@
+extern crate timely;
+extern crate timely_communication;
+
+use std::collections::HashMap;
+
+use timely_communication::Configuration;
+use timely::dataflow::operators::{ToStream, Input, Inspect, Operator, FrontierNotificator};
+use timely::dataflow::channels::pact::Pipeline;
+use timely::progress::timestamp::RootTimestamp;
+use timely::dataflow::Scope;
+
+fn main() {
+    timely::execute(Configuration::Thread, |root| {
+        root.scoped(|scope| {
+            (0u64..10).to_stream(scope)
+                .unary_frontier(Pipeline, "example", |mut builder| {
+                    let mut cap = Some(builder.get_cap(RootTimestamp::new(12)));
+                    move |(input, _frontier), output| {
+                        cap = None;
+                        while let Some((time, data)) = input.next() {
+                            output.session(&time).give_content(data);
+                        }
+                    }
+                }).inspect(|x| println!("{:?}", x));
+        });
+    }).unwrap();
+
+    timely::execute(Configuration::Thread, |root| {
+        root.scoped(|scope| {
+            (0u64..10).to_stream(scope)
+                .unary(Pipeline, "example", |mut builder| {
```

Comment:
That makes sense, I think I still kinda prefer the two nested closures: no need for Rc<RefCell<_>> and you get a reasonable spot where you can initialise local state or a user-space notificator, i.e.:

```
long().stream().composition()
  .that().spans().multiple().lines()
  .unary(exchange, "example", |cap| {
    let stash = Vec::new();
    let notificator = FrontierNotificator::new();
    move |input, output| {
      // logic ..
    }
  });
```

I tried to do 1. (Frontier in `InputHandle`) and it worked out pretty well. I made the `Option` private and only exposed a "method" that calls `.expect()` so that the user doesn't need to unwrap manually and the computation would still blow up if someone tries to use the frontier on an input that doesn't support it. Code here https://github.com/frankmcsherry/timely-dataflow/pull/39/commits/c7f768af1cd42c223097d07159de07f97e516cfb

2. (user-level `Notificator`) is already a thing (take a look at the examples in this PR): does it match what you had in mind?

I'm not sure I understand 3. Right now the capabilities are "internal"; how would capabilities associated with output(s) look like?